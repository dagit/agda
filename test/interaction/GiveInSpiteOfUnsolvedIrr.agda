{-# OPTIONS -v tc.conv.irr:20 #-}

module GiveInSpiteOfUnsolvedIrr where

postulate
  Level : Set
  lzero : Level
  lsuc  : (i : Level) → Level
  _⊔_   : Level -> Level -> Level

{-# BUILTIN LEVEL     Level #-}
{-# BUILTIN LEVELZERO lzero  #-}
{-# BUILTIN LEVELSUC  lsuc   #-}
{-# BUILTIN LEVELMAX  _⊔_ #-}

postulate
  .irrAxiom : ∀ {a}{A : Set a} → .A → A
{-# BUILTIN IRRAXIOM irrAxiom #-}

postulate
  _≡_ : {A : Set} → A → A → Set
  funExt : ∀ {A : Set}{B : A → Set}{f g : (x : A) → B x} →
    (∀ x → f x ≡ g x) → f ≡ g

record MC (M : Set) (op : M → M → M) : Set where
  field
    elem   : M
    fun    : M → M
    .match : fun ≡ op elem

unitM : ∀ {M : Set} {op : M → M → M} → MC M op
unitM {M = M} = record
  { elem = {!!}
  ; fun  = λ x → x
  ; match = {!funExt ?!}  -- try give here
  }

{-
No variable of type
{M : Set} {.op : M → M → M} → (λ x → x) ≡ .op ?0 was found in
scope.
-}

-- This declaration needs to stay to trigger the error!
compM : ∀ {M : Set} {op : M → M → M} → MC M op → MC M op → MC M op
compM = {!!}
