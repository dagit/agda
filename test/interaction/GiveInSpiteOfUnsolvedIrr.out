(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking GiveInSpiteOfUnsolvedIrr (GiveInSpiteOfUnsolvedIrr.agda).\n" t)
(agda2-verbose "instance search for solution of irrelevant meta\n_74\n:\n{M : Set} {.op : M → M → M} → (λ x → x) ≡ .op ?0\n")
(agda2-verbose "instance search for solution of irrelevant meta\n_74\n:\n{M : Set} {.op : M → M → M} → (λ x → x) ≡ .op ?0\n")
(agda2-verbose "instance search for solution of irrelevant meta\n_74\n:\n{M : Set} {.op : M → M → M} → (λ x → x) ≡ .op ?0\n")
(agda2-info-action "*Type-checking*" "Finished GiveInSpiteOfUnsolvedIrr.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : M\n?1 : (λ x → x) ≡ .op ?0\n?2 : MC .M .op → MC .M .op → MC .M .op\n" nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
(agda2-give-action 1 "funExt ?")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : M\n?2 : MC .M .op → MC .M .op → MC .M .op\n?3 : (x : M) → x ≡ .op ?0 x\n" nil)
((last . 1) . (agda2-goals-action '(0 3 2)))
